client/app/src/main/java/com/microsoft/research/karya/ui/registration/PhoneNumberFragment.kt

```diff
@@ -101,13 +118,14 @@ class PhoneNumberFragment : Fragment() {
             phoneNumberNextIv.isClickable = false
         }
     }
+
     /** On next click, hide keyboard. Send request to send OTP to the phone number */
     private fun handleNextClick() {
         baseActivity.hideKeyboard()
         WorkerInformation.phone_number = phoneNumberEt.text.toString()
-        startActivity(Intent(activity, SendOTPActivity::class.java))
-    }
 
+        startForResult.launch(Intent(activity, SendOTPActivity::class.java))
```

Comment:
`SendOTP` should be a repository method called from the activity instead.

Reply:
Yes, for now we have NetworkActivity that does that. It will be fixed once we use extensions to clear up NetworkActivity